### PR TITLE
wip

### PR DESCRIPTION
**Pull request by maliheh-deriv**
# Description 
Verify exchange rate for floating rate 

# cards 
https://app.clickup.com/t/20696747/P2PS-2460
https://app.clickup.com/t/20696747/P2PS-2423

### CODE REVIEW

**Review by markodonnell-deriv — CHANGES_REQUESTED** [review]
Custom commands need the need the c_ prefix as per the coding standards here: https://wikijs.deriv.cloud/en/Quality-Assurance/e2e-test-automation

I'm guessing certain commands are embedded within the P2P.js because they are generic (i.e. could be used by other tests) and the commands in createAdBuy.cy.js are specific to createAdBuy tests only?

@bilalmajid-deriv @meeyun-deriv . This is the standard I have proposed in the past, shall we add it to the new doc?

**Review by meeyun-deriv — COMMENTED** [review]
The test does not delete the created ad after running the test or it did not delete all existing ad before starting the test. So if the current test run failed in creating new ad, will the test pass as the account already has intended ad for validation?

File: cypress/e2e/wip/P2P/createAdBuy.cy.js [hunk 1]
```diff
@@ -1,29 +1,101 @@
...
+import {} from '../../../support/p2p'
 
-describe('QATEST-2414 - Create a Buy type Advert : Floating Rate', () => {
-  beforeEach(() => {
-    cy.c_login()
-    cy.c_visitResponsive('/appstore/traders-hub', 'small')
+
+let rate = 0.01
+let marketRate
+let ratecalculation
```
Comment (meeyun-deriv) [inline cypress/e2e/wip/P2P/createAdBuy.cy.js:7]: follow camelCase

**Review by meeyun-deriv — CHANGES_REQUESTED** [review]
The test does not delete the created ad after running the test or it did not delete all existing ad before starting the test. So if the current test run failed in creating new ad, will the test pass as the account already has intended ad for validation?

File: cypress/e2e/wip/P2P/createAdBuy.cy.js (hunk 1 above)
Comment (maliheh-deriv) [reply cypress/e2e/wip/P2P/createAdBuy.cy.js:7]: Regarding delete all ads , we have this [subtask](https://app.clickup.com/t/20696747/P2PS-2473) which I will be start working on it in another PR. they are all part of parent [card](https://app.clickup.com/t/20696747/P2PS-2423)

**Review by meeyun-deriv — APPROVED** [review]
LGTM

File: cypress/e2e/wip/P2P/createAdBuy.cy.js [hunk 2]
```diff
@@ -1,97 +1,25 @@
...
-    cy.findByPlaceholderText('Add').should('not.be.exist')
-    // Post ad 
-    cy.findByRole("button", { name: "Post ad" }).should("be.enabled").click()
-    cy.findByText("You've created an ad").should("be.visible")
-    cy.findByText("If the ad doesn't receive an order for 3 days, it will be deactivated.").should("be.visible")
-    cy.findByText("Don’t show this message again.").should("be.visible")
-    cy.findByRole("button", { name: "Ok" }).should("be.enabled").click()
+  it('should be able to create buy type advert and verify all fields and messages', () => {
+    cy.c_redirectToP2P()
+    cy.c_ClickMyAdTab()
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wip/P2P/createAdBuy.cy.js:13]: update to c_clickMyAdTab() to follow format cy.c_camelCase

File: cypress/e2e/wip/P2P/createAdBuy.cy.js [hunk 3]
```diff
@@ -1,97 +1,25 @@
 import '@testing-library/cypress/add-commands'
+import { } from '../../../support/p2p'
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wip/P2P/createAdBuy.cy.js:2]: can remove if not to be used

File: cypress/support/p2p.js [hunk 4]
```diff
@@ -1,3 +1,182 @@
...
+    if (match) {
+      cy.verifyExchangeRate
+      // Verify clicking plus button twice 
+      cy.get('#floating_rate_input_add').click({ force: true }).click({ force: true })
+      rate = rate + 0.02
+      cy.verifyExchangeRate
+      // // verify minus button once  
+      cy.get('#floating_rate_input_sub').click({ force: true })
+      rate = rate - 0.01
+      cy.verifyExchangeRate
```
Comment (bilalmajid-deriv) [inline cypress/support/p2p.js:84]: cy.verifyExchange needs to be updated to cy.c_verifyExchangeRate()

File: cypress/support/p2p.js [hunk 5]
```diff
@@ -1,3 +1,182 @@
...
+})
+
+Cypress.Commands.add('c_verifyCompletionOrderDropdown', () => {
+  cy.findByTestId('dt_dropdown_display').click()
+  cy.get('#3600').should("be.visible")
+  cy.get('#2700').should("be.visible")
+  cy.get('#1800').should("be.visible")
+  cy.get('#900').should("be.visible").click()
+  
+  // cy.xpath('//*[@id=900]').click()
```
Comment (bilalmajid-deriv) [inline cypress/support/p2p.js:114]: please remove commented code

File: cypress/support/p2p.js [hunk 6]
```diff
@@ -1,3 +1,182 @@
...
+    // else  {
+      cy.c_removeExistingAds()
+      cy.findByRole("button", { name: "Create new ad" }).should("be.visible").click()
+    } 
+
+  }) 
+})
+
+Cypress.Commands.add('c_ClickMyAdTab', () => {
+  cy.get('.notification__close-button').should("be.visible").click()
```
Comment (bilalmajid-deriv) [inline cypress/support/p2p.js:39]: we have a custom command for this to close notification headers as c_closeNotificationHeader() which can be reused

File: cypress/support/p2p.js [hunk 7]
```diff
@@ -1,3 +1,182 @@
...
+  cy.get("#dt_mobile_drawer_toggle").should("be.visible")
+  cy.get("#dt_mobile_drawer_toggle").click()
+  // click on cashier 
+  cy.get('.dc-mobile-drawer__body > :nth-child(4)').should("be.visible").click()
+  // click on P2P 
+  cy.findByText("Deriv P2P").should("be.visible").click()
+  // confirm warning message
+  cy.get('.dc-checkbox__box').should("be.visible").click()
+  cy.findByRole("button", { name: "Confirm" }).click()
+})
```
Comment (bilalmajid-deriv) [inline cypress/support/p2p.js:20]: we have two custom commands for this as well 

cy.c_navigateToDerivP2P()
cy.c_closeSafetyInstructions()

File: cypress/support/p2p.js [hunk 8]
```diff
@@ -1,3 +1,182 @@
...
+      throw new Error('Text does not match the expected pattern');
+    }
+
+  })
+})
+
+Cypress.Commands.add('c_verifyPostAd', () => {
+  cy.findByRole("button", { name: "Post ad" }).should("be.enabled").click()
+  cy.findByText("You've created an ad").should("be.visible")
+  // cy.findByText("If the ad doesn't receive an order for 3 days, it will be deactivated.").should("be.visible")
```
Comment (bilalmajid-deriv) [inline cypress/support/p2p.js:96]: comment can be removed if not needed.

**Review by bilalmajid-deriv — CHANGES_REQUESTED** [review]
Changes suggested.

Question:
As per standard decided commands are only going to be used if the code is redundant  and can be used in multiple test files are all these commands required?

File: cypress/e2e/wip/P2P/createAdBuy.cy.js (hunk 3 above)
Comment (maliheh-deriv) [reply cypress/e2e/wip/P2P/createAdBuy.cy.js:2]: All functions defined in the support/p2p and only called in here.

**Review by markodonnell-deriv — APPROVED** [review]
Looks good

File: cypress/support/p2p.js [hunk 9]
```diff
@@ -1,3 +1,165 @@
...
+
+Cypress.Commands.add('c_createNewAd', () => {
+  cy.findByTestId('dt_initial_loader').should('not.exist')
+  cy.get('body', { timeout: 10000 }).then((body) => {
+    // if (body.find('You have no ads').length > 0) {
+      if (body.find('.no-ads__message',{ timeout: 10000 }).length > 0){
+      cy.findByRole("button", { name: "Create new ad" }).should("be.visible").click()
+    }
+    else if (body.find('#toggle-my-ads',{ timeout: 10000 }).length > 0) {
+    // else  {
```
Comment (bilalmajid-deriv) [inline cypress/support/p2p.js:16]: sorry to have missed this could you please remove these comments on line 11 and 16 as well

**Review by bilalmajid-deriv — CHANGES_REQUESTED** [review]

**Comment by maliheh-deriv** [comment]
Next we will have createBuyAd for `fix rate`. That test will reuse these commands.

**Review by bilalmajid-deriv — APPROVED** [review]
LGTM

**Pull request merged**